### PR TITLE
Move to node20, release v3

### PR DESCRIPTION
**Pull request by jonny-puma**
Node 12 and 16 has reached end of life. Github is moving to node 20 spring 2024: https://github.blog/changelog/2023-09-22-github-actions-transitioning-from-node-16-to-node-20/

### CODE REVIEW

**Review by ohmyguigs — APPROVED** [review]
Nice

**Review by hamirmahal — COMMENTED** [review]
@denniskigen or @rschristian could either of you review this?

**Comment by hamirmahal** [comment]
```yml
        uses: preactjs/compressed-size-action@2.5.0
```

results in warnings in repositories like `prettier`.

**Comment by hamirmahal** [comment]
https://github.com/prettier/prettier/actions/runs/8629661553

**Comment by hamirmahal** [comment]
image

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "compressed-size-action",
-  "version": "2.5.0",
+  "version": "3.0.0",
```
Comment (hamirmahal) [inline package.json:3]: Is the major version bump here necessary? I'm genuinely asking. I don't know if using `node20` will break anything downstream for any consumers of this action.

**Comment by hamirmahal** [comment]
image

File: package.json (hunk 1 above)
Comment (hamirmahal) [inline package.json:3]: Is the major version bump here necessary? I'm genuinely asking. I don't know if using `node20` could potentially break anything downstream for any consumers of this action.

**Comment by rschristian** [comment]
@hamirmahal Make one comment, and edit it if you want to add more information. I, nor should anyone else, be pinged 7 times in the course of a few minutes for this.

**Comment by hamirmahal** [comment]
```yml
        uses: preactjs/compressed-size-action@2.5.0
```

results in warnings in repositories like `prettier`.

https://github.com/prettier/prettier/actions/runs/8629661553

image

File: package.json (hunk 1 above)
Comment (rschristian) [inline package.json:3]: Please revert this & the unrelated lock file change, we'll bump the version when releasing.

**Review by rschristian — APPROVED** [review]
Cheers
